conn.go

```diff
@@ -488,18 +488,27 @@ func (c *Conn) sendFrameWithReceipt(f *Frame) error {
 // will be received by this subscription. A subscription has a channel
 // on which the calling program can receive messages.
 func (c *Conn) Subscribe(destination string, ack AckMode) (*Subscription, error) {
+	return c.SubscribeWithHeaders(destination, ack, NewHeader())
+}
+
+func (c *Conn) SubscribeWithHeaders(destination string, ack AckMode, userDefined *Header) (*Subscription, error) {
```

Comment:
Nit: `userDefined` might not actually be defined by the user (eg: if invoked via `Subscribe`). I'd suggest naming something more descriptive like `headers` or `inputHeaders`?